pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -224,25 +263,51 @@ def newpkgcb(pkgId, name, arch):
         packages_pb.save()
         erratum_pb.save()
 
+        remote_url = self.new_url or self.remote.url
+
         with ProgressBar(message='Downloading Metadata Files') as metadata_pb:
             downloader = self.remote.get_downloader(
-                url=urljoin(self.remote.url, 'repodata/repomd.xml')
+                url=urljoin(remote_url, 'repodata/repomd.xml')
             )
             # TODO: decide how to distinguish between a mirror list and a normal repo
             result = await downloader.run()
             metadata_pb.increment()
 
+            if self.kickstart:
+                d_artifacts = []
+                for path, checksum in self.kickstart["download"]["images"].items():
+                    artifact = Artifact()
+
+                    if checksum:
```

Comment:
I'm not 100% sure, but can't this loop also be handled as L#279 being written as:   `artifact = Artifact(**checksum)` and this loop could be dropped?